test/unit-test/mqtt_agent_utest.c

```diff
@@ -74,21 +110,536 @@ int suiteTearDown( int numFailures )
+ * @param[in] pFuncName String of function name to print for error messages.
+ */
+static void invalidParamsTestFunc( MQTTStatus_t ( *FuncToTest )( MQTTAgentContext_t *, CommandInfo_t * ),
+                                   const char * pFuncName )
+{
+    MQTTAgentContext_t agentContext = { 0 };
+    MQTTStatus_t mqttStatus;
+    CommandInfo_t commandInfo = { 0 };
+
+    setupAgentContext( &agentContext );
+
+    /* Test for NULL parameters. */
+    mqttStatus = ( FuncToTest ) ( &agentContext, NULL );
+    TEST_ASSERT_EQUAL_MESSAGE( MQTTBadParameter, mqttStatus, pFuncName );
+
+    mqttStatus = ( FuncToTest ) ( NULL, &commandInfo );
+    TEST_ASSERT_EQUAL_MESSAGE( MQTTBadParameter, mqttStatus, pFuncName );
+
+    /* Various NULL parameters for the agent interface. */
+    agentContext.agentInterface.send = NULL;
+    mqttStatus = ( FuncToTest ) ( &agentContext, &commandInfo );
+    TEST_ASSERT_EQUAL_MESSAGE( MQTTBadParameter, mqttStatus, pFuncName );
+
+    agentContext.agentInterface.send = stubSend;
+    agentContext.agentInterface.recv = NULL;
+    mqttStatus = ( FuncToTest ) ( &agentContext, &commandInfo );
+    TEST_ASSERT_EQUAL_MESSAGE( MQTTBadParameter, mqttStatus, pFuncName );
+
+    agentContext.agentInterface.recv = stubReceive;
+    agentContext.agentInterface.getCommand = NULL;
+    mqttStatus = ( FuncToTest ) ( &agentContext, &commandInfo );
+    TEST_ASSERT_EQUAL_MESSAGE( MQTTBadParameter, mqttStatus, pFuncName );
+
+    agentContext.agentInterface.getCommand = stubGetCommand;
+    agentContext.agentInterface.releaseCommand = NULL;
+    mqttStatus = ( FuncToTest ) ( &agentContext, &commandInfo );
+    TEST_ASSERT_EQUAL_MESSAGE( MQTTBadParameter, mqttStatus, pFuncName );
+
+    agentContext.agentInterface.releaseCommand = stubReleaseCommand;
+    agentContext.agentInterface.pMsgCtx = NULL;
+    mqttStatus = ( FuncToTest ) ( &agentContext, &commandInfo );
+    TEST_ASSERT_EQUAL_MESSAGE( MQTTBadParameter, mqttStatus, pFuncName );
+
+    agentContext.agentInterface.pMsgCtx = &globalMessageContext;
+    /* Invalid packet ID to indicate uninitialized context. */
+    agentContext.mqttContext.nextPacketId = MQTT_PACKET_ID_INVALID;
+    mqttStatus = ( FuncToTest ) ( &agentContext, &commandInfo );
+    TEST_ASSERT_EQUAL_MESSAGE( MQTTBadParameter, mqttStatus, pFuncName );
+}
+
+/* ========================================================================== */
+
+/**
+ * @brief Test that MQTTAgent_Init is able to update the context object correctly.
+ */
+
+void test_MQTTAgent_Init_Happy_Path( void )
+{
 }
 
 /**
  * @brief Test that any NULL parameter causes MQTTAgent_Init to return MQTTBadParameter.
  */
 void test_MQTT_Init_Invalid_Params( void )
 {
-    
+}
+
+/* ========================================================================== */
+
+/**
+ * @brief Test error cases when a command cannot be obtained or sent
+ * for API functions that create commands.
+ */
+void test_MQTTAgent_API_Create_Command_Creation_Error( void )
```

Comment:
The name is misleading. This test should be broken into two:

1. MQTTAgent_Ping_Queue_Full [or whatever the reason for out of memory is].
2. MQTTAgent_Ping_Send_To_Queue_Failed 

The idea is that a test should test only one thing and it  must only fail when that one thing is false.